### PR TITLE
LFA Feedback 2 - API Endpoints 

### PR DESCRIPTION
**Pull request by jptuyisenge**
## What does this PR do?
This PR fulfill thee LFA's feedback about the API endpoints

### Steps to reproduce
1. Create an environment file to keep a secret key of the token
2. Put the validation in the middleware
3. Use the only ES6
4. Make sure to use the proper status codes
5. Remove unnecessary conditions in the controllers

## Expected
1. The secret will be kept in the environment file
2. Having Validation in the middleware
3. Usage of ES6 throughout the application 
4. Well usage of HTTP status code
5. Reviewing all the conditions in the controllers

## Actual
1. The secret key for the token well is not protected
2. Unordered usage of validation in the middleware
3. Mixage of Javascript ES5 and ES6
4. Improper usage of status code
5. Unnecessary conditions in the controller
## How should this be manually tested?
N/A
## Any background context you want to provide?
N/A
## What are the relevant pivotal tracker stories?
[#168898263](https://www.pivotaltracker.com/story/show/168898263)

## Screenshots (if appropriate)
N/A

## Questions:
N/A

### CODE REVIEW

File: UI/Js/employee.js [hunk 1]
```diff
@@ -147,62 +145,62 @@ let closeComment = document.getElementsByClassName('close')[5];
 let closeQuery = document.getElementsByClassName('close')[6];
 // let closeNewArticle = document.getElementsByClassName("close")[0];
```
Comment (NkFab) [inline UI/Js/employee.js]: Remove this commented line

File: UI/Js/employee.js [hunk 2]
```diff
@@ -147,62 +145,62 @@ let closeComment = document.getElementsByClassName('close')[5];
...
 
-closeNewArticle.onclick = function () {
+closeNewArticle.onclick = () => {
   let successBox = document.querySelector('.success-msg-new-article');
   successBox.style.display = 'none';
   // successBox.setAttribute('class','easyout');
```
Comment (NkFab) [inline UI/Js/employee.js]: Remove this commented line

File: server/app.js [hunk 3]
```diff
@@ -9,17 +9,22 @@ app.use(express.json());
...
+});
+
+app.use((_req, _res, next) => {
   const error = new Error('Bad url ');
   error.status = 404;
   next(error);
 });
 
-app.use((error, req, res, next) => {
+app.use((error, _req, res) => {
```
Comment (NkFab) [inline server/app.js]: Why did you change `req` to `_req`

File: server/helpers/token.js [hunk 4]
```diff
@@ -1,8 +1,7 @@
 import jwt from 'jsonwebtoken';
-import tokenKeys from '../config/keys';
 
 const Token = (payLoad) => {
-  const Newtoken = jwt.sign({ payLoad }, tokenKeys, {
+  const Newtoken = jwt.sign({ payLoad }, process.env.tokenKeys, {
```
Comment (NkFab) [inline server/helpers/token.js]: It's best practice to name your variables in the dotenv file in a capital case format and use an underscore to separate words in this instance `TOKEN_KEY`.

File: server/middleware/tokenVerification.js [hunk 5]
```diff
@@ -18,7 +19,7 @@ const isLoggedin = (req, res, next) => {
   }
 
   try {
-    const decodedToken = jwt.verify(token, keys);
+    const decodedToken = jwt.verify(token, process.env.tokenKeys);
```
Comment (NkFab) [inline server/middleware/tokenVerification.js]: It's best practice to name your variables in the dotenv file in a capital case format and use an underscore to separate words in this instance TOKEN_KEY.

File: server/routes/comments.js [hunk 6]
```diff
@@ -3,9 +3,10 @@ import comment from '../controllers/comments';
 import loggedInUser from '../middleware/loggedInUser';
 import tokenVerification from '../middleware/tokenVerification';
 import paramCheck from '../middleware/parameterCheck';
+import Validate from '../middleware/validation';
 
 const router = express.Router();
-router.post('/commentes/:id', tokenVerification, paramCheck.parameterCheck, comment.createComment);
+router.post('/commentes/:id', tokenVerification, paramCheck.parameterCheck, Validate.addComment, comment.createComment);
```
Comment (NkFab) [inline server/routes/comments.js]: I think you wanted to write comments instead of commentes. Change wherever you have this word.

File: UI/Js/employee.js (hunk 1 above)
Comment (jptuyisenge) [reply UI/Js/employee.js]: Removed

File: UI/Js/employee.js (hunk 2 above)
Comment (jptuyisenge) [reply UI/Js/employee.js]: removed

File: server/helpers/token.js (hunk 4 above)
Comment (jptuyisenge) [reply server/helpers/token.js]: Done

File: server/middleware/tokenVerification.js (hunk 5 above)
Comment (jptuyisenge) [reply server/middleware/tokenVerification.js]: Done

File: server/routes/comments.js (hunk 6 above)
Comment (jptuyisenge) [reply server/routes/comments.js]: Replaced

File: server/app.js (hunk 3 above)
Comment (jptuyisenge) [reply server/app.js]: Es lint showed me the error on that parameter. In trying to fix that with using esLint hints I choose `prefix all unused declaration with '_'  where possible`,  
And then after this happened and has no impact on the application workflow.
